### PR TITLE
mysqldef: support parsing scientific notation

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: mysqldef/decimal.go [hunk 1]
```diff
@@ -199,17 +199,28 @@ func NewDecimalFromUint(value uint64, exp int32) Decimal {
 //
 func ParseDecimal(value string) (Decimal, error) {
 	var intString string
-	var exp int32
+	var exp = int32(0)
```
Comment (qiuyesuifeng) [inline mysqldef/decimal.go]: Origin code is also int32(0)?

File: mysqldef/decimal.go [hunk 2]
```diff
@@ -199,17 +199,28 @@ func NewDecimalFromUint(value uint64, exp int32) Decimal {
...
+
+	n := strings.IndexAny(value, "eE")
+	if n > 0 {
```
Comment (qiuyesuifeng) [inline mysqldef/decimal.go]: How about "e1"?

Comment (siddontang) [inline mysqldef/decimal.go]: invalid format

**Pull request merged**